### PR TITLE
feat(canary): Adds list command for canary configs.

### PR DESCRIPTION
**Pull request by jtk54**
https://github.com/spinnaker/spinnaker/issues/4857. 

FYI all the unit tests are bjorked from the Gate version check in client.go, apparently they are not running. You can stub in the `/version` call like I did for this unit test to fix that issue. We seem to repeat ourselves a bunch in the unit tests, might be worth a refactor to keep things clean.

### CODE REVIEW

**Comment by jtk54** [comment]
Nested `canary-config` under `canary` and added the `application` flag.

**Comment by duftler** [comment]
image

File: cmd/canary/canary-config/list_test.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+// to direct requests to. Responds with a 200 and a well-formed canaryConfig list.
+func testGateCanaryConfigListSuccess() *httptest.Server {
+	mux := http.NewServeMux()
+	mux.Handle(
+		"/v2/canaryConfig",
+		http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			fmt.Fprintln(w, strings.TrimSpace(canaryConfigListJson))
+		}))
+	// Return a Gate version.
+	mux.Handle("/version", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```
Comment (louisjimenez) [inline cmd/canary/canary-config/list_test.go]: If you rebase and replace lines 101 and 116 with `mux := util.TestGateMuxWithVersionHandler()` this Handle call for version can be removed, also the one on 123.

**Comment by jtk54** [comment]
Updated and addressed comments.

**Pull request merged**